pkg/services/sqlstore/active_node.go

```diff
@@ -85,6 +91,10 @@ func InsertActiveNodeHeartbeat(cmd *m.SaveActiveNodeCommand) error {
 				sqlog.Debug(errmsg, "error", err)
 				return errors.New(errmsg + ": " + err.Error())
 			}
+			if cmd.ParticipantLimit > 0 && partID == int64(cmd.ParticipantLimit) {
```

Comment:
It is correct. 
If participantLimit is 0 or less, that means all nodes are participating. Otherwise, if new partID is = participantLimit, that means the limit is reached. Give attention to what is returned.